pkg/targets/adapter/googlecloudstoragetarget/adapter.go

```diff
@@ -104,15 +106,15 @@ func (a *googlecloudstorageAdapter) instertArbitraryObject(ctx context.Context,
 }
 
 func (a *googlecloudstorageAdapter) insertObject(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
-	ep := &EventPayload{}
-	if err := event.DataAs(ep); err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing, err, nil)
+	d, err := json.Marshal(event)
+	if err != nil {
+		return a.replier.Error(&event, "error marshalling CloudEvent", err, nil)
 	}
 
-	obj := a.bucket.Object(ep.FileName)
+	obj := a.bucket.Object(event.ID() + ".json")
 	w := obj.NewWriter(ctx)
 
-	if _, err := w.Write(ep.Data); err != nil {
+	if _, err := w.Write(d); err != nil {
 		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
```

Comment:
```suggestion
		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "writing the data")
```

Reply:
We should avoid performing unrelated changes in this PR.

If we want to improve the usage of the replier, let's just open a "Improve the usage of the replier" PR.